### PR TITLE
Generalize Salesforce Event Sources

### PR DESCRIPTION
**Pull request by jverce**
* Add event source logic for:
  * New objects
  * Updated objects
  * Deleted objects
* Add user input prop to allow users to select the type of sObject for
  which new events should be monitored

This PR is in support of issue #532

### CODE REVIEW

File: components/salesforce/sources/new-object/new-object.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+const common = require("../../common");
+
+const EVENT_SOURCE_NAME = "New Object (Instant)";
+
+module.exports = {
+  ...common,
+  name: EVENT_SOURCE_NAME,
+  key: "salesforce-new-object",
+  description: "Triggers when a new object is created",
```
Comment (dylburger) [inline components/salesforce/sources/new-object/new-object.js:11]: Can you rephrase to the "Emits an event..." format?

**Review by dylburger — APPROVED** [review]

File: components/salesforce/sources/new-object/new-object.js (hunk 1 above)
Comment (jverce) [reply components/salesforce/sources/new-object/new-object.js]: Done!
